libqtile/widget/wlan.py

```diff
@@ -41,9 +41,30 @@ def get_status(interface_name):
     return essid, quality
 
 
+def eth_get_status(interface_name):
+    # From kernel docs https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-class-net:
+    # Carrier:
+    # == =====================
+    # 0  physical link is down
+    # 1  physical link is up
+    # == =====================
+    # Operstate
+    # Possible values are:
+    # "unknown", "notpresent", "down", "lowerlayerdown", "testing",
+    # "dormant", "up".
+    try:
+        with open("/sys/class/net/" + interface_name + "/carrier") as fcarrier:
+            cable_status = fcarrier.readline().rstrip("\n")
+        with open("/sys/class/net/" + interface_name + "/operstate") as fstate:
+            link_status = fstate.readline().rstrip("\n")
+    except FileNotFoundError:
+        return None, None
+    return cable_status, link_status
```

Comment:
> From a user's POV, what's the difference between the carrier and operstate values? Is it that carrier is "cable is plugged or unplugged" and operstate is "connected or disconnected to network"?

The carrier shows 0 or 1 if the cable is unplugged or plugged. The operstate show the ability to send and receive packages, not exactly if it's connected to the internet. For example, if it can send and receive packages to other computers in the local network, it'll also display "up". I can improve the documentation if needed to make it clearer.

>Would it be reasonable/possible to make the helper (and I guess format options) independent of the type of the interface i.e. ethernet vs wlan? There are other types (e.g. wg0 for wireguard) that seem to have the same sysfs interface.

I think it's possible. Be aware that /sys/class/net only shows information about the device itself. It's hard to say, for example, if there is an IP address associated or if it's the preferred net device. This is all handled by Network Manager and, AFAIK, you can get this info by talking with Networking Manager (using dbus or something), not in the sysfs. Therefore, maybe iwlib uses this "advanced" feature and is not only sysfs (my guess).